FeedSimulator/src/main/scala/bythebay/FeederActor.scala

```diff
@@ -0,0 +1,53 @@
+package bythebay
+
+import akka.actor.{Props, Actor, ActorLogging}
+import bythebay.FeederActor.Initialize
+import bythebay.FeederActor.SendNextLine
+import bythebay.FeederActor.ShutDown
+import kafka.producer.KeyedMessage
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+
+/**
+ * This keeps the file handle open and just reads on line at fixed time ticks.
+ * Not the most efficient implementation, but it is the easiest.
+ */
+class FeederActor extends Actor with ActorLogging with FeederExtensionActor{
+
+  var counter = 0
+
+  implicit val executionContext = context.system.dispatcher
+
+  val feederTick = context.system.scheduler.schedule(Duration.Zero, 100.millis, self, SendNextLine)
+
+  var ratingsDataSet:Iterator[String] = Iterator.empty
```

Comment:
Hmm... since the name of the file is always read from immutable config, I would suggest just making this a `lazy val`, and the actor can be simplified a lot ... no need for `Initialize` or the check to see if it is empty.